honeycombio/data_source_query.go

```diff
@@ -284,3 +284,71 @@ func validateQueryJSON(validators ...querySpecValidateDiagFunc) schema.SchemaVal
 		return diagnostics
 	}
 }
+
+func extractFilters(d *schema.ResourceData) ([]honeycombio.FilterSpec, error) {
+	filterSet := d.Get("filter").(*schema.Set)
+	hashFn := filterSet.F
+	filters := make([]honeycombio.FilterSpec, filterSet.Len())
+	for i, schemaFilter := range filterSet.List() {
+		honeyFilter, err := extractFilter(d, hashFn(schemaFilter))
+		if err != nil {
+			return nil, err
+		}
+		filters[i] = honeyFilter
+	}
+	return filters, nil
+}
```

Comment:
It took me a little while to figure out how to reference nested properties.
Since we're using a set instead of a list you need to find the generated hash, then you can access nested objects/properties via `d.Get('filter.<id>')`/`d.Get('filter.<id>.value')`